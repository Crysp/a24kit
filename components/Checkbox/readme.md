```js
<div>
    <div><Checkbox label="Обычный чекбокс"/></div><br/>
    <div><Checkbox label="Большой чекбокс" large/></div>
</div>
```

#### Выбранный чекбокс
```js
<div>
    <div><Checkbox label="Обычный чекбокс" isChecked/></div><br/>
    <div><Checkbox label="Большой чекбокс" large isChecked/></div>
</div>
```

#### Неактивный чекбокс
```js
<div>
    <div><Checkbox label="Обычный чекбокс" disabled/></div><br/>
    <div><Checkbox label="Большой чекбокс" large disabled/></div>
</div>
```

#### Неактивный выбранный чекбокс
```js
<div>
    <div><Checkbox label="Обычный чекбокс" isChecked disabled/></div><br/>
    <div><Checkbox label="Большой чекбокс" large isChecked disabled/></div>
</div>
```

#### Цвета
```js
<div>
    <div><Checkbox label="Синий чекбокс" isChecked/></div><br/>
    <div><Checkbox label="Зеленый чекбокс" color="green" isChecked/></div><br/>
    <div><Checkbox label="Темно-синий чекбокс" color="deepBlue" isChecked/></div>
</div>
```