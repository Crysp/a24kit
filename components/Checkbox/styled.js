import styled, { css } from 'styled-components';
import { Wrapper as MarkWrapper } from './components/Mark/styled';
import { Color } from '../config';


function borderColor(props) {
    if (props.disabled) {
        return css`border-color: ${Color.Pale};`;
    }
    switch (props.color) {
        case 'green':
            return css`border-color: ${Color.Green};`;
        case 'deepBlue':
            return css`border-color: ${Color.DeepBlue};`;
        default:
            return css`border-color: ${Color.Blue};`;
    }
}

export const Wrapper = styled.div`
    display: inline-block;
    cursor: pointer;
    user-select: none;
    line-height: 50%;
    white-space: nowrap;
    pointer-events: ${props => (props.disabled ? 'none' : 'inherit')};
    
    &:hover ${MarkWrapper} {
        ${borderColor}
    }
`;

export const Label = styled.div`
    display: inline-block;
    margin-left: 12px;
    font: 400 14px "Open Sans", sans-serif;
    line-height: 20px;
    vertical-align: top;
    white-space: normal;
    color: ${props => (props.disabled ? Color.Gray : 'inherit')};
`;
