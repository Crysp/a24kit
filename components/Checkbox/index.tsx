import React from 'react';
import Mark from './components/Mark';
import { Wrapper, Label } from './styled';

export interface Props {
    color: 'blue' | 'green' | 'deepBlue'
    isChecked: boolean
    label: React.ReactNode
    large: boolean
    disabled: boolean
    onCheck(): void
}

export default class Checkbox extends React.Component<Props> {
    static defaultProps = {
        color: 'blue',
        isChecked: false,
        label: '',
        large: false,
        disabled: false,
        onCheck() {},
    };

    onCheck = () => {
        const { disabled, onCheck } = this.props;
        if (disabled) return;
        onCheck();
    };

    render() {
        const {
            label,
            disabled,
            ...other
        } = this.props;
        delete other.onCheck;
        delete other.large;
        const hasLabel = (typeof label === 'string' && label.length > 0) || label;
        return (
            <Wrapper disabled={disabled} onClick={this.onCheck} {...other}>
                <Mark {...this.props} />
                {hasLabel && <Label disabled={disabled}>{label}</Label>}
            </Wrapper>
        );
    }
}
