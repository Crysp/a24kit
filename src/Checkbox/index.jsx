import React from 'react';
import PropTypes from 'prop-types';
import Mark from './components/Mark';
import { Wrapper, Label } from './styled';


/**
 * @develop
 */
export default class Checkbox extends React.Component {
    static propTypes = {
        /**
         * Цвет
         */
        color: PropTypes.oneOf(['blue', 'green', 'deepBlue']),
        /**
         * Выбран ли элемент
         */
        isChecked: PropTypes.bool,
        /**
         * Текстовая подпись
         */
        label: PropTypes.node,
        /**
         * Увеличенный размер
         */
        large: PropTypes.bool,
        /**
         * Неактивный
         */
        disabled: PropTypes.bool,
        /**
         * Обработчик выбора
         */
        onCheck: PropTypes.func,
    };

    static defaultProps = {
        color: 'blue',
        isChecked: false,
        label: '',
        large: false,
        disabled: false,
        onCheck() {},
    };

    onCheck = () => {
        const { disabled, onCheck } = this.props;
        if (disabled) return;
        onCheck();
    };

    render() {
        const {
            label,
            disabled,
            ...other
        } = this.props;
        delete other.onCheck;
        delete other.large;
        const hasLabel = (typeof label === 'string' && label.length > 0) || label;
        return (
            <Wrapper disabled={disabled} onClick={this.onCheck} {...other}>
                <Mark {...this.props} />
                {hasLabel && <Label disabled={disabled}>{label}</Label>}
            </Wrapper>
        );
    }
}
