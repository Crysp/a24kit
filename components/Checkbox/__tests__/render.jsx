import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../index';


describe('Верстка компонента <Checkbox>', () => {
    it('Дефолтная', () => {
        const tree = renderer.create(<Checkbox />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Синего цвета', () => {
        const tree = renderer.create(<Checkbox color="blue" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Зеленого цвета', () => {
        const tree = renderer.create(<Checkbox color="green" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Темносинего цвета', () => {
        const tree = renderer.create(<Checkbox color="deepBlue" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Ставится галочка', () => {
        const tree = renderer.create(<Checkbox isChecked />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Выводится лейбл', () => {
        const tree = renderer.create(<Checkbox label="Label" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Увеличенный', () => {
        const tree = renderer.create(<Checkbox large />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Неактивный', () => {
        const tree = renderer.create(<Checkbox disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
