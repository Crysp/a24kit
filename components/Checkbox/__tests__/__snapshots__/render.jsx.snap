// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <Checkbox> Выводится лейбл 1`] = `
<div
  className="sc-ifAKCX gkzOrl"
  color="blue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hwFftZ"
    disabled={false}
  >
    <div
      className="sc-htpNat jAdlEj"
      color="blue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-EHOje hijERh"
    disabled={false}
  >
    Label
  </div>
</div>
`;

exports[`Верстка компонента <Checkbox> Дефолтная 1`] = `
<div
  className="sc-ifAKCX gkzOrl"
  color="blue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hwFftZ"
    disabled={false}
  >
    <div
      className="sc-htpNat jAdlEj"
      color="blue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Зеленого цвета 1`] = `
<div
  className="sc-ifAKCX ipiSZX"
  color="green"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hwFftZ"
    disabled={false}
  >
    <div
      className="sc-htpNat PhFqh"
      color="green"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Неактивный 1`] = `
<div
  className="sc-ifAKCX epkeYp"
  color="blue"
  disabled={true}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH jwtyQG"
    disabled={true}
  >
    <div
      className="sc-htpNat kjsDPm"
      color="blue"
      disabled={true}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Синего цвета 1`] = `
<div
  className="sc-ifAKCX gkzOrl"
  color="blue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hwFftZ"
    disabled={false}
  >
    <div
      className="sc-htpNat jAdlEj"
      color="blue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Ставится галочка 1`] = `
<div
  className="sc-ifAKCX gkzOrl"
  color="blue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH jEjhxV"
    disabled={false}
  >
    <div
      className="sc-htpNat gqgabp"
      color="blue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Темносинего цвета 1`] = `
<div
  className="sc-ifAKCX kxdDuv"
  color="deepBlue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hwFftZ"
    disabled={false}
  >
    <div
      className="sc-htpNat fWwtbt"
      color="deepBlue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb keOSFw sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;

exports[`Верстка компонента <Checkbox> Увеличенный 1`] = `
<div
  className="sc-ifAKCX gkzOrl"
  color="blue"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH cdMfyU"
    disabled={false}
  >
    <div
      className="sc-htpNat jAdlEj"
      color="blue"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="sc-bxivhb jIZPyy sc-bdVaJa dTKkTG"
        focusable="false"
        size="auto"
        style={Object {}}
        viewBox="0 0 8 7"
      >
        <path
          d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
        />
      </svg>
    </div>
  </div>
  
</div>
`;
